examples/vision/super_resolution_sub_pixel.py

```diff
@@ -0,0 +1,513 @@
+paper and trained with relatively small dataset
+[BSDS500](http://www.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/BSR/BSR_bsds500.tgz).
+"""
+
+"""
+## Setup
+"""
+
+import tensorflow as tf
+
+"""
+## Load data: BSDS500 dataset
+
+### Download data
+
+We use Keras built-in util `keras.utils.get_file` to get the dataset.
+"""
+
+dataset_url = "http://www.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/BSR/BSR_bsds500.tgz"
+data_dir = tf.keras.utils.get_file(origin=dataset_url, fname="BSR", untar=True)
+
+"""
+Check the `data_dir` and build data path.
+"""
+
+import os
+
+BSDS500 = os.path.join(data_dir, "BSDS500/data")
+print(BSDS500)
+
+"""
+Prepare train and test dataset path and check the number of train, test and validation
+samples.
+"""
+
+dataset = os.path.join(BSDS500, "images")
+train_path = os.path.join(dataset, "train")
+test_path = os.path.join(dataset, "test")
+valid_path = os.path.join(dataset, "val")
+
+input_img_paths = sorted(
+    [
+        os.path.join(train_path, fname)
+        for fname in os.listdir(train_path)
+        if fname.endswith(".jpg")
+    ]
+)
+
+test_img_paths = sorted(
+    [
+        os.path.join(test_path, fname)
+        for fname in os.listdir(test_path)
+        if fname.endswith(".jpg")
+    ]
+)
+
+valid_img_paths = sorted(
+    [
+        os.path.join(valid_path, fname)
+        for fname in os.listdir(valid_path)
+        if fname.endswith(".jpg")
+    ]
+)
+
+print("Number of train samples: ", len(input_img_paths))
+print("Number of test samples: ", len(test_img_paths))
+print("Number of validation samples: ", len(valid_img_paths))
+
+"""
+## Visualiza the data
+"""
+
+from IPython.display import display
+from tensorflow.keras.preprocessing.image import load_img
+
+display(load_img(input_img_paths[0]))
+
+"""
+## Crop and resize images.
+
+Since we need to upscale images and we have to process the input data format. First, you
+need to define crop size and calculate the acutal crop size by following functions. For
+input data, after we get the crop image and blur it with `bicubic` method. For target
+data, we only do the crop operation.
+"""
+
+import PIL
+
+
+def calculate_crop_size(crop_size, upscale_factor):
+    return crop_size - (crop_size % upscale_factor)
+
+
+def transform_input(input, crop_size, upscale_factor):
+    input = input.crop((0, 0, crop_size, crop_size))
+    crop_size = crop_size // upscale_factor
+    return input.resize((crop_size, crop_size), PIL.Image.BICUBIC)
+
+
+def transform_target(input, crop_size):
+    input = input.crop((0, 0, crop_size, crop_size))
+    return input
+
+
+"""
+## Visualize input and target data in YCbCr color space.
+"""
+
+"""
+We luminance in `YCbCr` colour space for images because the changes in that space are
+easy to see.
+"""
+
+from tensorflow.keras.preprocessing.image import array_to_img
+from tensorflow.keras.preprocessing.image import img_to_array
+import numpy as np
```

Comment:
Let's use `tf.xxx` APIs directly if possible, rather than importing the symbols directly.

Reply:
I checked other examples on keras.io, they didn't use in that way, they usually import like this.
```
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
```
I think I should keep the consistency.

Reply:
+1